lstchain/tools/lstchain_fit_intensity_scan.py

```diff
@@ -0,0 +1,364 @@
+
+import os
+import numpy as np
+import h5py
+from scipy.optimize import curve_fit
+from traitlets import List, Int, Dict,Float
+from pathlib import Path
+from matplotlib.backends.backend_pdf import PdfPages
+from matplotlib import pyplot as plt
+from ctapipe.core import Tool, traits
+from matplotlib.ticker import MaxNLocator
+from ctapipe.containers import (
+    FlatFieldContainer,
+    WaveformCalibrationContainer,
+    PedestalContainer,
+)
+from ctapipe.io.hdf5tableio import HDF5TableReader
+from ctapipe_io_lst import constants
+
+__all__ = [
+    'FitIntensityScan'
+]
+
+MIN_N_RUNS = 5
+
+class FitIntensityScan(Tool):
+    """
+     Tool that generates a HDF5 file with the results of the fit
+     of the signal of an intensity scan (filter scan in the case of LST), this is useful to estimate the
+     quadratic noise term to include in the standard F-factor formula
+
+     To be run with
+     lstchain_fit_intensity_scan --config config.json
+
+     """
+
+    name = "FitFilterScan"
+    description = "Tool to fit an intensity scan"
+
+    signal_range = List(
+        [[1500, 14000],[200, 14000]],
+        help='Signal range to include in the fit for [HG,LG] (camera median in [ADC])'
+    ).tag(config=True)
+
+    gain_channels = List(
+        [0,1],
+        help='Gain channel to process (HG=0, LG=1)'
+    ).tag(config=True)
+
+    sub_run = Int(
+        0,
+        help='Sub run number to process'
+    ).tag(config=True)
+
+    run_list = List(
+        help='List of runs',
+    ).tag(config=True)
+
+    input_dir = traits.Path(
+        directory_ok=True,
+        help='directory with the input files',
+    ).tag(config=True)
+
+    input_prefix = traits.Unicode(
+        default_value="calibration",
+        help='Prefix to select calibration files to fit',
+    ).tag(config=True)
+
+    output_path = traits.Path(
+        directory_ok=False, default_value="filter_scan_fit.h5",
+        help='Path the output hdf5 file',
+    ).tag(config=True)
+
+    plot_path = traits.Path(
+        directory_ok=False, default_value="filter_scan_fit.pdf",
+        help='Path to pdf file with check plots',
+    ).tag(config=True)
+
+    fit_initialization = List(
+        [[100.0, 0.001], [6.0, 0.001]],
+        help='Fit parameters initalization [gain (ADC/pe), B term] for HG and LG'
+    ).tag(config=True)
+
+    fractional_variance_error = Float(
+        0.02,
+        help='Costant fractional error assumed for the y fit coordinate (variance)'
```

Comment:
Costant ==> Constant
However why do you use the constant uncertainty? You can compute it easily from the variance itself and number of values.

Reply:
I prefer to leave the possibility to add some systematic contribution. By the way, which formula would you use?

Reply:
sigma_error = sigma / sqrt(2N - 2), where N is the number of measurements from which sigma is computed. 
see e.g.: https://math.stackexchange.com/questions/1259383/calculating-uncertainty-in-standard-deviation
I tested it with a simple toy MC and it works perfect well for a simple gaussian distribution and > 10 measurements. 0.02 that you are using would correspond to ~1250 events. I guess this is more or less in the correct ballpark for the statistics that you are using, but if you have large differences in statistics between different filter wheel positions the fit might be more stable/precise if you use proper uncertainties